tests/unit/utils/decorators/collection-action-test.ts

```diff
@@ -0,0 +1,43 @@
+import { collectionAction } from 'ember-api-actions/decorators';
+import DS from 'ember-data';
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+import setupPretender from '../../../helpers/setup-pretender';
+
+class Fruit extends DS.Model {
+  @collectionAction({ path: 'ripenEverything' })
+  public ripenAll;
```

Comment:
<img width="587" alt="Bildschirmfoto 2020-04-09 um 19 45 07" src="https://user-images.githubusercontent.com/141300/78924773-a7fab880-7a9a-11ea-9304-842229f7b6af.png">

I'm seeing quite a few TypeScript warnings/errors in these tests 🤔

Reply:
I'll see if I can do anything about that

Reply:
I wonder if it has anything to do with the TypeScript version? The one being used in this library is pretty old... 🤔

Reply:
I got all but these two cleaned up

<img width="1248" alt="Screen Shot 2020-04-09 at 2 20 59 PM" src="https://user-images.githubusercontent.com/1645881/78927769-6bfd2e80-7a6d-11ea-8184-56e1a8885481.png">

I don't understand why to do about that error... The method returns the type signature that TypeScript wants for a property decorator, as far as I can tell

https://www.typescriptlang.org/docs/handbook/decorators.html#property-decorators

Reply:
Never mind -- I got those fixed up too.

I think the confusion stems from the difference between how TypeScript and Babel see decorators; since Babel actually transpiles the decorator, you get the third `descriptor` parameter, but the TypeScript support for decorators does _not_ expect that third parameter to be provided -- and you _must_ mark the return-type of the decorator itself to be explicitly `any`.